public/app/plugins/panel/alertlist/UnifiedAlertList.tsx

```diff
@@ -90,74 +91,91 @@ export function UnifiedAlertList(props: PanelProps<UnifiedAlertListOptions>) {
   useEffect(() => {
     if (props.options.groupMode === GroupMode.Default) {
       dispatch(
-        fetchAllPromAndRulerRulesAction(false, {
-          limitAlerts: limitInstances ? INSTANCES_DISPLAY_LIMIT : undefined,
-          matcher: matcherList,
-          state: stateList,
-        })
+        fetchAllPromAndRulerRulesAction(
```

Comment:
I wonder whether instead of adding a new parameter to the `fetchAllPromAndRulerRulesAction`, we could use `fetchPromAndRulerRulesAction` (this one fetches rules for only one data source) when we know that only one data source has been selected.
It's a bit strange to pass one data source to a function named `fetchAll`

Reply:
good point, I will take a look